DOC: Fixup to AnchoredArtist examples in the gallery

madphysicist opened the pull request:
Response to issue #11092. Instead of deleting the user gallery example called `anchored_bbox01`, I made it a non-duplicate. It did not make sense to remove it entirely given that there are perfectly reasonable `anchored_bbox` examples up to `04`. I did remove the clearly obsolete `misc/anchored_artists.html` and renamed `axes_grid1/simple_anchored_artists.html` to `axes_grid1/anchored_artists.html`. Hopefully the changes I made to the comments and docstrings in the example came out as an improvement.

ImportanceOfBeingErnest commented:
See comment on original issue #11092. In short: they are non-duplicates. So at most, I'd recommend adding explanations above the respective examples providing the context and link to the other one(s).

madphysicist commented on examples/userdemo/anchored_box01.py:
```diff
@@ -10,8 +10,9 @@
-at = AnchoredText("Figure 1a",
-                  prop=dict(size=15), frameon=True, loc=2)
+# loc=8 is equivalent to loc='lower center'
+at = AnchoredText("Here is some\nsample text",
```
@ImportanceOfBeingErnest @jklymak Would this be considered an acceptable change?
1. Modified the string to have a newline, because that (A) makes it less redundant, and (B) shows off the capabilities a little better
2. Changed (and documented) the `loc` parameter.

madphysicist commented on examples/axes_grid1/simple_anchored_artists.py:
```diff
@@ -56,13 +74,12 @@ def draw_sizebar(ax):
     ax.add_artist(asb)
 
 
-if 1:
```
Not sure why that was here in the first place...

madphysicist commented on examples/misc/anchored_artists.py:
```diff
@@ -56,48 +69,57 @@ def __init__(self, transform, width, height, angle, loc,
 
+    ax.add_artist(ae)
 
-if __name__ == "__main__":
```
Well, OK. I did delete this line, and made the same set of small functions as in the similar file to highlight the parallels.

madphysicist commented on examples/misc/anchored_artists.py:
```diff
@@ -56,48 +69,57 @@ def __init__(self, transform, width, height, angle, loc,
+draw_ellipse(ax)
+draw_sizebar(ax)
 
-    plt.draw()
```
And I got rid of this line. It doesn't do anything useful if you call `show` next.

madphysicist commented:
@ImportanceOfBeingErnest @jklymak. I think you will find my changes much more acceptable now. I moved the code around in the non-toolkit file to line up better with the yes-toolkit one. Besides that, I added a bunch of comments to clarify why there are two files there to begin with. I've documented all the other changes I made through review comments here, all of which I think are legitimate cleanup.

ImportanceOfBeingErnest commented:
In the simple *Anchored Box01* example, I would not change the text actually. This example is meant to be the simplest case for an anchored text (as opposed to a text positionned by coordinates). But you can get rid of the loc=2 and replace it with "upper left". As said it would be good if the code shown in the example and the user guide are equal. 

```
at = AnchoredText("Figure 1a",
                  prop=dict(size=15),
                  frameon=True,
                  loc="upper left")
```

Maybe instead you can use the *misc_anchored_artists* example to display a two-line text, if that is important for you?

As said, I think there should be links between the files. E.g. to link from the `misc_anchored_artists` example to 
to the `axes_grid1_simple_anchored_artists` you'd use

    :ref:`sphx_glr_gallery_axes_grid1_simple_anchored_artists.py`

madphysicist commented:
What exactly would be the purpose of <code>:ref:\`sphx_glr_gallery_axes_grid1_simple_anchored_artists.py`</code>? The text would just appear in a comment, or am I missing something?

ImportanceOfBeingErnest commented:
The purpose would be that one can click on the link and land on the example page. Did you try that? Is it not working?

madphysicist commented:
@ImportanceOfBeingErnest I actually don't know how to build the docs locally. Will figure it out and let you know.

ImportanceOfBeingErnest commented:
Building the docs can be a little bit cumbersome the first time you try it, especially on windows. Of course it's ideal to test locally first before committing anything, but you may also rely on the CI tests here on the server for building the docs. It might be a bit overkill going through all the hassle for testing if a single link it working correctly.

madphysicist commented:
I'm doing this on a Linux box that's not doing much else at the moment. It sounds like the kind of thing I'd want to try once for fun just to get an idea, so if you have a pointer to some instructions, I'd be happy to do it.

jklymak commented:
https://matplotlib.org/devel/index.html

ImportanceOfBeingErnest commented:
In general, the instructions at [writing documentation](https://matplotlib.org/devel/documenting_mpl.html#installing-dependencies) apply. In addition make sure not to use Sphinx 1.7.3. Have an up to date latex available, including xetex. I think that's it for linux. (You never know what else can go wrong until you try it.)

madphysicist commented:
@ImportanceOfBeingErnest Tried it, ran into the issue of no such package `matplotlib`. I would prefer to build the docs in place without installing mpl into my current environment. I suspect that this can be done by setting up the PYTHONPATH to point to my source folder, but I was hoping there is a more "appoved" solution.

ImportanceOfBeingErnest commented:
Mhh not sure if I understand this. Did you build matplotlib in a source folder and then installed it in a python environment? In that case inside that environment you should be able to do `import matplotlib` - that's the whole point of it, right? Otherwise you couldn't even test any of the changes to the code.

madphysicist commented:
@ImportanceOfBeingErnest. I was basically looking for `python setup.py develop`, or more optimally `pip install -e .`. My lack of knowledge of the build system is apalling. I am used to numpy's `runtests.py`, which builds everything in place and does not properly install it as far as I am aware.

madphysicist commented:
I've made all the requested changes. I did not understand how the comments in the examples were being rendered because most of the examples I have seen don't have proper docstrings. Thanks for bearing with me on this one. I've learned a lot about building and installing matplotlib, as well as how to work with the docs.

jklymak commented:
Thanks for the effort on this @madphysicist
